src/gui/DatabaseWidget.cpp

```diff
@@ -1272,8 +1272,11 @@ void DatabaseWidget::reloadDatabaseFile()
         }
     } else {
         m_messageWidget->showMessage(
-            tr("Could not open the new database file while attempting to autoreload this database."),
+            tr("Could not open the new database file while attempting to autoreload this database.")
+                    .append("\n").append(file.errorString()),
             MessageWidget::Error);
+        // Mark db as modified since existing data may differ from file or file was deleted
+        m_db->modified();
```

Comment:
This should be called `setModified()`.

Reply:
This directly calls the modified signal so that both the tab widget and dbwidget get updated as "modified". There is no other way to do that in the current architecture. I will add a //hack note.

Reply:
Ok.